### ISSUE TITLE
Make Inline / TextElement inherit from InputElement to enable pointer/keyboard input handling

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#10186]
**Is your feature request related to a problem? Please describe.**

In order to implement Hyperlink inline pointer / keyboard input is required.

**Describe the solution you'd like**

Make Inline / TextElement inherit from InputElement to enable pointer/keyboard input handling

**Describe alternatives you've considered**

Embedding TextBlock's controls?

**Additional context**

https://github.com/AvaloniaUI/Avalonia/blob/d39b927266e18670f230b0fd16d58074f65dc512/src/Avalonia.Controls/Documents/Inline.cs#L11

https://github.com/AvaloniaUI/Avalonia/blob/d39b927266e18670f230b0fd16d58074f65dc512/src/Avalonia.Controls/Documents/TextElement.cs#L9

https://github.com/AvaloniaUI/Avalonia/blob/d39b927266e18670f230b0fd16d58074f65dc512/src/Avalonia.Base/Input/InputElement.cs#L21

**Comment by Gillibald** [comment]
The concrete implementation `InputElement` should not be required to handle input in some inline. Inline itself is not a Visual so this will not be possible. Instead only `IInputElement` should be required.

**Comment by wieslawsoltes** [comment]
> So how did you end up implementing this on your side?

https://github.com/wieslawsoltes/HackerNews/blob/7f4593ac1b2ec92a3ad935c5a959c59c11a479ff/HackerNews/Controls/Hyperlink.cs#L14

**Comment by Gillibald** [comment]
I don't think this issue is solved. You are adding a styled button to the text layout. This issue is about making TextRun's hit testable for commands etc.

**Comment by AndrewDRX** [comment]
Just noting to offer a potential workaround for some for the time being until a `Hyperlink:Inline` is available...

My primary desire for a `Hyperlink:Inline` vs the `HyperlinkButton:Control` is that the `Inline` would ideally have proper inline wrapping. For the derived `Control`, wrapping will only occur if an explicit `Width` or `MaxWidth` is set.

As a workaround, I found logic similar to this to be helpful to set that explicit `MaxWidth` based on the width changes of the parent `TextBlock` control:
```csharp
var parent = hyperlink.FindAncestorOfType<TextBlock>();
if (parent is not null)
{
    parent.SizeChanged += (sender, e) => hyperlink.MaxWidth = e.NewSize.Width;
}
```

This will need to be done after the `HyperlinkButton` is added to the tree.

It also has the caveat that the `HyperlinkButton` will display inline only if the full width can be displayed inline, otherwise the `HyperlinkButton` will start on the next line, and then will wrap itself to subsequent lines until it is fully displayed, and then the next sibling `Inline` will begin on the line below that.

For example:
```
[Short Run][Short HyperlinkButton][Short Run]

[Short Run]
[Long HyperlinkButton]
[Short Run]
```

However, that behavior is preferrable to having the `HyperlinkButton` expand beyond the bounds of the containing `TextBlock` w/out wrapping.
